src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -210,11 +225,54 @@ export class UserService {
     });
     if (!userData) throw new NotFoundException('User not found');
 
+    if (dto.mail) {
+      await this.tfaService.checkVerification(userData, ip, TfaLevel.BASIC);
+
+      const user = await this.userRepo.findOneBy({ userData: { mail: dto.mail } });
+      if (user) throw new BadRequestException('Mail already in use');
```

Comment:
Should we maybe check on this point if a merge for the userDatas would be possible? 🤔 
We have a entity method in userData `checkIfMergePossibleWith` we could use for this